src/cache/cache.service.ts

```diff
@@ -1,40 +1,46 @@
 import { Inject, Injectable } from '@nestjs/common';
-import { readFile, writeFile, unlink, mkdir } from 'fs/promises';
+import { readFile, writeFile, unlink, mkdir, stat } from 'fs/promises';
 import { join } from 'path';
 import {
   CACHE_DIR,
   CACHE_DEFAULT_VALUE,
   CACHE_FILE_NAME,
+  CACHE_BATCH_SIZE,
 } from './cache.constants';
 import { ProviderService } from 'provider';
 
 @Injectable()
-export class CacheService<T extends unknown> {
+export class CacheService<
+  H extends unknown,
+  D extends unknown,
+  T extends { headers: H; data: D[] } = { headers: H; data: D[] },
+> {
   constructor(
     private providerService: ProviderService,
     @Inject(CACHE_DIR) private cacheDir: string,
     @Inject(CACHE_FILE_NAME) private cacheFile: string,
+    @Inject(CACHE_BATCH_SIZE) private cacheBatchSize: number,
     @Inject(CACHE_DEFAULT_VALUE) private cacheDefaultValue: T,
   ) {}
 
   private cache: T | null = null;
 
   public async getCache(): Promise<T> {
     if (!this.cache) {
-      this.cache = await this.getCacheFromFile();
+      this.cache = await this.getCacheFromFiles();
     }
 
     return this.cache;
   }
 
   public async setCache(cache: T): Promise<void> {
     this.cache = cache;
-    return await this.saveCacheToFile();
+    return await this.saveCacheToFiles();
```

Comment:
Maybe add removal of cache files before actual saving of new files?

Reply:
done in `saveCacheToFiles` method